src/include/migraphx/op/reshape.hpp

```diff
@@ -64,45 +65,83 @@ struct reshape
 
     std::string name() const { return "reshape"; }
 
-    shape dyn_compute_shape(shape s0) const
+    // Assumes that the shape from the `dims` attribute will be valid at run-time.
+    // Makes no checks for the validity of the `dims` attribute for the given input shape.
+    shape dyn_1arg_compute_shape(shape s0) const
     {
-        auto dyn_dims = s0.dyn_dims();
-        // track number of fixed elements in input and output
-        std::size_t num_dims_ele = 1;
-        std::size_t num_dd_ele   = 1;
-        for(std::size_t i = 0; i < dyn_dims.size(); ++i)
+        auto input_dyn_dims = s0.dyn_dims();
+        bool has_negative_dim_attr = false;
+        // construct output dynamic shape from dims attribute
+        std::vector<shape::dynamic_dimension> output_dyn_dims(dims.size());
+        for(int i = 0; i < dims.size(); ++i)
         {
-            if(dyn_dims[i].is_fixed())
+            int64_t d = dims.at(i);
+            if(d == 0)
+            {
+                output_dyn_dims.at(i) = input_dyn_dims.at(i);
+            }
+            else if(d == -1)
             {
-                num_dims_ele *= dims[i];
-                num_dd_ele *= dyn_dims[i].min;
+                has_negative_dim_attr = true;
+                output_dyn_dims.at(i) = {1, 1};
             }
             else
             {
-                if(dims[i] != 0 and dims[i] != -1)
-                {
-                    MIGRAPHX_THROW(
-                        "Reshape: Non-fixed dynamic_dimension doesn't match with 0 or -1 "
-                        "output dimension");
-                }
+                std::size_t u_dim     = d;
+                output_dyn_dims.at(i) = {u_dim, u_dim};
             }
         }
-        if(num_dims_ele != num_dd_ele)
+
+        if(has_negative_dim_attr)
         {
-            MIGRAPHX_THROW("Reshape: Number of fixed elements must match. Input: " +
-                           std::to_string(num_dd_ele) + " Output: " + std::to_string(num_dims_ele));
+            // comparing the -1 dimension against the other dimensions
+            auto neg_dim_num = std::distance(this->dims.begin(),
```

Comment:
Isn't the negative dimension only allowed as the last dimension?

Reply:
No, it can be any of the dimensions. But there can only be one -1 dimension.